File: tests/codegen/snapshots/typescript/oneof_typename.ts
type OneOfTypenameTestResultOneOfTypenamePerson = {
    name: string
    age: number
}

type OneOfTypenameTestResult = {
    // alias for one_of_typename
    alias: OneOfTypenameTestResultOneOfTypenamePerson
}

type OneOfInput = {
    a: string,
    b: never
} | {
    a: never,
    b: string
}

type OneOfTypenameTestVariables = {
    value: OneOfInput
}

File: tests/codegen/snapshots/python/oneof_typename.py
from typing import Union

class OneOfTypenameTestResultOneOfTypenamePerson:
    # typename: Person
    name: str
    age: int

class OneOfTypenameTestResult:
    # alias for one_of_typename
    alias: OneOfTypenameTestResultOneOfTypenamePerson

class OneOfInputA:
    a: str

class OneOfInputB:
    b: str

OneOfInput = Union[OneOfInputA,OneOfInputB]

class OneOfTypenameTestVariables:
    value: OneOfInput
